### PR TITLE
Replace search controller with react-query

### PR DESCRIPTION
**Pull request by kurtisassad**
## Link to Issue
Closes: #8290

## Description of Changes
- Removes search controller and replaces with react-query

## Test Plan
- Create a thread, go to the 3 dots (more options) click the edit collaborators button. Search for profiles.

image

### CODE REVIEW

**Review by Israellund — APPROVED** [review]
LGTM

File: packages/commonwealth/client/scripts/controllers/server/search.ts [hunk 1]
```diff
@@ -1,361 +0,0 @@
...
-import type { SearchParams } from '../../models/SearchQuery';
-import SearchQuery, { SearchScope } from '../../models/SearchQuery';
-
-const SEARCH_PREVIEW_SIZE = 6;
-const SEARCH_PAGE_SIZE = 10;
-const SEARCH_HISTORY_KEY = 'COMMONWEALTH_SEARCH_HISTORY';
-const SEARCH_HISTORY_SIZE = 10;
-
-class SearchController {
-  private _store: SearchStore = new SearchStore();
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/controllers/server/search.ts:16]: please remove SearchStore.ts

File: packages/commonwealth/client/scripts/controllers/server/search.ts [hunk 2]
```diff
@@ -1,361 +0,0 @@
-import { SearchContentType } from '@hicommonwealth/shared';
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/controllers/server/search.ts:1]: SearchContentType seems to be unused so please delete it

**Review by masvelio — APPROVED** [review]
let's remove dead code from the repository. Thanks for taking first shot!

**Comment by kurtisassad** [comment]
> Edit collaborators work but then when I want to display it, it shows all anonymous

Bug was unrelated but fixed in commit above this comment

**Pull request merged**